pkg/nimble/README.ipv6-over-ble.md

```diff
@@ -0,0 +1,157 @@
+between Linux and RIOT (using GNRC and NimBLE).
+
+## Prerequisites
+
+You need a BLE (NimBLE) capable RIOT board (currently only `nrf52`-based boards
+are supported), and a Linux host with at least Bluetooth 4.0 capabilities.
+
+Any Linux distribution should do fine, however this guide was tested with the
+following:
+- Mint 18.3
+- Kernel 4.15.0
+- bluez 5.37
+- [optional] radvd 2.11
+
+
+## Preparing the RIOT node
+
+Simply flash the `examples/gnrc_networking` application to your RIOT device.
+
+Once the firmware is running, you can verify it by typing
+
+    ble info
+
+This should give you some information about your device's BLE address and
+status information about open connections. It should look similar to the
+following:
+
+    Own Address: UU:VV:WW:XX:YY:ZZ -> [FE80::UUVV:WWFF:FEXX:YYZZ]
+     Free slots: 3/3
+    Advertising: no
+       Contexts:
+    [ 0] state: 0x8000 - unused
+    [ 1] state: 0x8000 - unused
+    [ 2] state: 0x8000 - unused
+
+Once this is working, you need to tell the node to accept incoming connections
+and start advertising itself:
+
+    ble adv
+
+Thats it, the node can now be seen and connected to from Linux.
+
+
+## Establishing a connection from Linux
+
+### Preparing Linux
+
+First of all, the Linux kernel needs to be configured to use 6lowpan over BLE.
+This needs to be done only once per session and consists of the following steps:
+
+    # Log in as a root user (IMPORTANT, as the following steps will not work
+    # using explicit `sudo` for each step...)
+    sudo su
+
+    # Mount debugfs file system, already the case on many distributions
+    mount -t debugfs none /sys/kernel/debug
+
+    # Load 6LoWPAN module
+    modprobe bluetooth_6lowpan
+
+    # Enable the bluetooth 6lowpan module.
+    echo 1 > /sys/kernel/debug/bluetooth/6lowpan_enable
+
+Next, we should verify that our Bluetooth device is configured properly:
+
+    # Look for available HCI devices.
+    hciconfig
+
+This should show us some information about the available Blutooth devices. If no
+device is listed here, something is wrong...
+
+
+### Connecting to the RIOT node
+
+When connecting to the RIOT device, you need to know the BLE address of the
+device. There are two ways to find out:
+
+a) use `ble info` or `ifconfig` on the RIOT device directly and copy the address
+
+b) use `bluetoothctl` on Linux to scan for the device. Once `bluetoothctl` has
+   started, issue `scan on` to start scanning. The default name for the RIOT
+   device is set to `RIOT-GNRC`, so you should see it pop up.
+
+c) use `hcitool lescan` on the Linux host (older tool, predecessor of
+   `bluetoothctl`...)
+
+Once you have the address, you simply connect Linux to RIOT using the following
+command:
+
+    # Put your device address here...
+    # Note: the 2 after the address denotes a BLE public random address, default
+    #       used by `nimble_netif`
+    echo "connect UU:VV:WW:XX:YY:ZZ 2" > /sys/kernel/debug/bluetooth/6lowpan_control
```

Comment:
I had the problem several times with the soft device as well, that I miss the 2 (even with the comment above) because I do not copy-paste this part as the address is different anyway. I'm not sure however how to make this more obvious :-/.

Reply:
I know, but IMHO this is clearly stated here and this should suffice.

Reply:
I know, just a potential pitfall, I will bet I won't be the last to fall into ;-). But if you have no idea how to improve it, let's keep it as is for now...